### PR TITLE
[ONNX] Added translation for int4/uint4 constants/inputs

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Added support for accepting int4/uint4 as constant/input

### Tickets:
 - 139934

### CODE REVIEW

File: src/frontends/onnx/tests/models/int4_const.prototxt [hunk 1]
```diff
@@ -0,0 +1,41 @@
+ir_version: 10
```
Comment (rkazants) [inline src/frontends/onnx/tests/models/int4_const.prototxt:1]: may be it is better to add example models with Add operation (parameter and Const)?

**Review by rkazants — APPROVED** [review]

File: src/frontends/onnx/tests/models/int4_const.prototxt (hunk 1 above)
Comment (rkazants) [reply src/frontends/onnx/tests/models/int4_const.prototxt:1]: I propose to create such test and disable it for now, referring CVS ticket requesting CPU support

File: src/frontends/onnx/frontend/src/core/tensor.hpp [hunk 2]
```diff
@@ -260,8 +270,22 @@ class Tensor {
...
-            constant = std::make_shared<ov::op::v0::Constant>(type, m_shape, data);
+            switch (type) {
+            case ov::element::i4:
+            case ov::element::u4:
+                constant = std::make_shared<ov::op::v0::Constant>(type, m_shape, data.data());
+                break;
+            default:
+                constant = std::make_shared<ov::op::v0::Constant>(type, m_shape, data);
+                break;
+            }
```
Comment (praasz) [inline src/frontends/onnx/frontend/src/core/tensor.hpp:288]:
```suggestion
            constant = std::make_shared<ov::op::v0::Constant>(type, m_shape, data.data());
```
As number elements in `data` is same as in Constant it should work in all cases.

File: src/frontends/onnx/frontend/src/core/tensor.hpp [hunk 3]
```diff
@@ -260,8 +270,22 @@ class Tensor {
         std::shared_ptr<ov::op::v0::Constant> constant{nullptr};
         auto data = get_data<T>();
         auto data_size = data.size();
```
Comment (praasz) [inline src/frontends/onnx/frontend/src/core/tensor.hpp:272]: Consider rename it to `element_count` ?

File: src/frontends/onnx/frontend/src/core/tensor.hpp [hunk 4]
```diff
@@ -260,8 +270,22 @@ class Tensor {
...
+        switch (type) {
```
Comment (praasz) [inline src/frontends/onnx/frontend/src/core/tensor.hpp:273]:
```suggestion
        ov::element::is_nibble_type(type) {
```
This helper is dev API `src/core/dev_api/openvino/core/type/element_iterator.hpp`

**Review by andrei-kochin — APPROVED** [review]

**Pull request merged**